supervisor/host/network.py

```diff
@@ -121,6 +124,16 @@ async def apply_changes(self, interface: Interface) -> None:
         """Apply Interface changes to host."""
         inet = self.sys_dbus.network.interfaces.get(interface.name)
 
+        # Handle issue with disabling IPv6 with old network manager
+        if (
+            interface.ipv6.method == InterfaceMethod.DISABLED
+            and self.sys_dbus.network.version < MIN_DISABLE_IPV6_NM_VERSION
+        ):
+            interface.ipv6.method = InterfaceMethod.LINK_LOCAL
+            _LOGGER.warning(
+                "Method 'disable' could not be used, using 'link-local' instead"
+            )
+
```

Comment:
Can we move that code into the payload generator?

Reply:
Yes, but why?